src/main/java/com/wb/service/WeatherService.java

```diff
@@ -0,0 +1,66 @@
+package com.wb.service;
+
+import com.wb.payload.Weather;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.AsyncResult;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.concurrent.Future;
+
+@Service
+public class WeatherService {
+
+	private static final Logger logger = LoggerFactory.getLogger(WeatherService.class);
+	private final RestTemplate restTemplate;
+
+	@Value("${open.weather.appid}")
+	private String APP_ID;
+
+	@Value("${open.weather.app.url}")
+	private String APP_URL;
+
+	public WeatherService(RestTemplateBuilder restTemplateBuilder) {
+		this.restTemplate = new RestTemplate();
+	}
+
+	@Cacheable(cacheNames = "weather", key = "#cityCountry")
+	public Weather fetchWeather(String cityCountry) throws RestClientException {
+		logger.info("Sync : Looking up " + cityCountry);
+		String url = String.format(APP_URL + "?q=%s&appid=%s", cityCountry, APP_ID);
+		return restTemplate.getForObject(url, Weather.class);
+	}
+
+	@Cacheable(cacheNames = "weather", key = "#cityCountryTime")
+	public Weather fetchWeatherByTime(String cityCountry, Long startTime, Long endTime) throws RestClientException {
+		logger.info("Sync : Looking up " + cityCountry);
+		String url = String.format(APP_URL + "?q=%s&appid=%s&type=hour&start=%s&end=%s", cityCountry, APP_ID, startTime,
```

Comment:
nie sklejaj url samemu. `RestTemplate` umie to zrobić za Ciebie: https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html#getForEntity-java.lang.String-java.lang.Class-java.util.Map-